starknet_py/net/full_node_client.py

```diff
@@ -65,7 +65,12 @@ def __init__(
         """
         self.url = node_url
         self._client = RpcHttpClient(url=node_url, session=session)
-        self._chain = chain_from_network(net, chain)
+
+        try:
+            self._chain = chain_from_network(net, chain)
+        except ValueError:
+            self._chain = None
```

Comment:
Why can't we just check if chain was provided and only then use chain_from_network?

Reply:
Because if someone is using testnet or mainnet there is no need to provide a chain and still have a chain property correctly set

Reply:
I can add this:
if chain is not None and net in ["mainnent", "testnet"]:
   self.chain = chain_form_network

Reply:
I'd rather do this.